### PR TITLE
Fix: Truncate long month names in research card button to prevent overflow (Issue #354)

### LINKED ISSUE #354: Truncate long month names in research card button to prevent overflow
# User Story

As a user viewing research cards, I want month names to display consistently without breaking the card layout so that the interface remains clean and readable across all months.

# Description

The month label inside the research card button currently overflows the button’s border when the month name is long (e.g., "September"). This results in layout distortion on the frontend. The month names should be shortened to a consistent abbreviated format (e.g., "Sep.") to maintain UI stability.

Image

# Proposed Flow

- [ ] Detect long month names in the research card button
- [ ] Apply standardized abbreviation rules (e.g., Jan., Feb., Mar., Sep.)
- [ ] Ensure abbreviated month names no longer overflow the button
- [ ] Verify consistent visual alignment across all card variations

# Reference

- Current research card layout in the frontend where the month label is rendered within the button component.

**Notes:**

- Ensure the abbreviation logic is applied at the data formatting layer or the UI rendering layer, depending on architecture conventions.
- Add regression tests for all months to prevent future overflow issues.

### PR DESCRIPTION
**Pull request by MichaelLeonardi**
<h3>Summary</h3>
<p>
This PR improves the visual presentation of date ranges displayed in
<strong>On-going Research</strong> cards on the Assessment page by standardizing
date formatting to prevent text overflow and layout breakage.
</p>

<h3>Purpose</h3>
<p>
Long month names (e.g. <em>September</em>) caused the research card date range
section to overflow its container, resulting in inconsistent card heights
and reduced readability, especially on smaller screens.
</p>
<p>
This change ensures date ranges remain compact, readable, and visually
consistent across all devices.
</p>

<h3>Behaviour (UI Changes)</h3>

<strong>Before</strong>
<ul>
  <li>Date ranges used full month names</li>
  <li>Text could overflow or wrap awkwardly inside research cards</li>
  <li>Cards appeared uneven and visually cluttered</li>
</ul>

<strong>After</strong>
<ul>
  <li>Month names are abbreviated (e.g. <code>Sep</code> instead of <code>September</code>)</li>
  <li>Date ranges fit cleanly within the card layout</li>
  <li>Research cards maintain consistent height and alignment</li>
  <li>Improved readability without changing user interaction flow</li>
</ul>

<h3>Implementation Notes</h3>
<ul>
  <li>Introduced a standardized <code>formatDateRange()</code> utility for research card date display</li>
  <li>Formatting adapts based on same/different month and year cases</li>
  <li>Redundant month/year values are removed when applicable</li>
  <li>Basic validation added to safely handle missing or invalid date values</li>
</ul>

<h3>Testing</h3>
<ul>
  <li>Verified on <strong>Assessment Centre → On-going Research</strong> section</li>
  <li>Confirmed correct rendering across multiple research entries with different date ranges</li>
</ul>

<p><strong>Fixes #354</strong></p>

## Summary by CodeRabbit

* **New Features**
  * Improved date range display for research items. Date ranges now format intelligently based on whether the start and end dates share the same year or month, providing clearer and more intuitive presentation.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/app/assessments/page.tsx [hunk 1]
```diff
@@ -42,6 +42,35 @@ const dateFormat = (date: string) => {
   return format(parseISO(date), 'dd MMMM yyyy');
 };
 
+const formatDateRange = (start: string, end: string) => {
+  if (!start || !end) return '';
+
+  const startDate = parseISO(start);
+  const endDate = parseISO(end);
+
```
[the remark this answers is not in the document]
Comment (lamurian) [reply src/app/assessments/page.tsx]: Wdyt of this suggestion @MichaelLeonardi? Additionally, I agree that the `formatDateRange` function is a general date utility. Thus, we can move it to `src/utils/dateUtils.ts` (create a new file).
